app/controllers/games_controller.rb

```diff
@@ -18,20 +14,143 @@ def show
   end
 
   def update
-    if game.valid? && game.white_player_id != game_params[:black_player_id]
+    if white_leaves_as_only_player?
+      close_game
+    elsif black_player_exists?
+      if black_player_leaves_without_moving?
+        update_game_for_white_after_black_leaves if current_player.id == game.white_player_id
+        
+        leave_game_open if current_player.id == game.black_player_id
+      elsif both_players_have_moved?
+        black_player = game_params[:black_player_id]
+        white_player = game_params[:white_player_id]
+        
+        if black_player.eql? ""
+          game.update_attribute(:winning_player_id, game.white_player_id)
+          
+          if current_player.id == game.white_player_id
+            set_game_winner_for_remaining_player(game.black_player_id)
+          else
+            set_game_winner_for_leaving_player(game.white_player_id)
+          end
+        end
+        
+        if white_player.eql? ""
+          game.update_attribute(:winning_player_id, game.black_player_id)
+          
+          if current_player.id == game.black_player_id
+            set_game_winner_for_remaining_player(game.white_player_id)
+          else
+            set_game_winner_for_leaving_player(game.black_player_id)
+          end
+        end
+      else
+        cancel_game
+      end
+    elsif players_are_unique?
       game.update_attributes game_params
       redirect_to game
     end
   end
 
   private
 
-  def game
-    @game ||= Game.where(id: params[:id]).last
-  end
+    def game
+      @game ||= Game.where(id: params[:id]).last
+    end
 
-  def game_params
-    params.require(:game).permit(:white_player_id, :black_player_id)
-  end
+    def game_params
+      params.require(:game).permit(:white_player_id, :black_player_id)
+    end
+    
+    def players_are_unique?
+      ( game.valid? && ( game.white_player_id != game_params[:black_player_id] ) ) ? true : false
+    end
+    
+    def black_player_exists?
+      game.black_player_id ? true : false
+    end
+    
+    def white_player_leaving?
+      game_params[:white_player_id].eql?("") ? true : false
+    end
+    
+    def black_player_leaving?
+      game_params[:black_player_id].eql?("") ? true : false
+    end
+    
+    def white_has_moved?
+      game.pieces.where(color: 'white', has_moved?: true).any? ? true : false
+    end
+    
+    def black_has_moved?
+      game.pieces.where(color: 'black', has_moved?: true).any? ? true : false
+    end
+  
+    def white_leaves_as_only_player?
+      ( white_player_leaving? && game.black_player_id.nil? ) ? true : false
+    end
+    
+    def black_player_leaves_without_moving?
+      black_has_moved? ? false : true if black_player_leaving?
+    end
+    
+    def both_players_have_moved?
+      ( white_has_moved? && black_has_moved? ) ? true : false
+    end
```

Comment:
Ref. Lines 66-100
Almost all these methods have unnecessary ternary operators included. The conditions in these methods should return a boolean value even without the ternary operators. Did you come across an issue where you felt you needed the ternary?